### PR TITLE
Recurring job start just after minute start

### PR DESCRIPTION
**Pull request by chanow816**
Signed-off-by: Chris Chien <chris.chien@suse.com>
From https://github.com/longhorn/longhorn/issues/3233, we have some flaky recurring job cases in daily run.

Most cases fail reason was timing issue. Longhorn recurring job started at  beginning of a minute, but our test cases will set recurring job directly (maybe at 55th second), then do some test steps then do sleep **whole** minute and check result in the end

Modified `create_recurring_jobs` and reduce sleep time in test cases.

### CODE REVIEW

**Comment by jenting** [comment]
>  Longhorn recurring job started at beginning of a minute, but our test cases will set recurring job directly 

Nice catch!

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -4326,6 +4327,12 @@ def update_setting(client, name, value):
 
 
 def create_recurring_jobs(client, recurring_jobs):
+    for i in range(60):
```
Comment (innobead) [inline manager/integration/tests/common.py:4330]: Why not just wait for the rest of seconds in a minute?

Something like

```python
time.sleep(60 - datetime.utcnow().second)
```

File: manager/integration/tests/test_recurring_job.py [hunk 2]
```diff
@@ -421,7 +421,7 @@ def test_recurring_job_in_storageclass(set_random_backupstore, client, core_api,
     for pod_name in pod_names:
         write_pod_volume_random_data(core_api, pod_name, volume_data_path, 2)
 
-    time.sleep(150)  # 2.5 minutes
+    time.sleep(140)  # 2.5 minutes - 10 secends
```
Comment (innobead) [inline manager/integration/tests/test_recurring_job.py:424]: If you already made the change in create_recurring_jobs, why need to change the sleep time? (including the all changes in this module)

**Review by innobead — CHANGES_REQUESTED** [review]

**Comment by innobead** [comment]
> > Longhorn recurring job started at beginning of a minute, but our test cases will set recurring job directly
> 
> Nice catch!

Indeed 👍

**Comment by chanow816** [comment]
Hi @innobead ,

The most flaky scenario was we set recurring job retain count to 2, scheduled each minute, after 1 whole minute passed we check backup should equal 1, however somehow the back count equal 2 (Because we checked too late, second retain already completed), this caused by:
- Actions between set recurring job and check backup count need time.
- We set recurring at end of minutes , like 55th second 

So I try to set recurrent job just after minute start and reduce sleep time in each flaky case, thank you.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@chanow816 plz backport to 1.2.x and 1.1.3. Thanks.

**Pull request merged**